src/connectionController.ts

```diff
@@ -192,8 +239,11 @@ export default class ConnectionController {
           'e.g. mongodb+srv://username:password@cluster0.mongodb.net/admin',
         prompt: 'Enter your connection string (SRV or standard)',
         validateInput: (uri: string) => {
-          if (!Connection.isURI(uri)) {
-            return 'MongoDB connection strings begin with "mongodb://" or "mongodb+srv://"';
+          try {
+            // eslint-disable-next-line no-new
+            new ConnectionString(uri);
```

Comment:
I like this change, but I'm wondering if the old one, although not as good of a validation check, might give a user unsure what to enter in something more information to help their case.
Mostly if they input something like a url but is invalid they'd get an `Invalid connection string` error message which doesn't hint at what might go there. https://github.com/mongodb-js/mongodb-connection-string-url/blob/main/src/index.ts#L106
Maybe something for that package instead of here. We do have a placeholder here that should give those users something to go off of so this may be a non-issue.

Reply:
@addaleax

Reply:
I can definitely see the concern here – if you want to keep the previous check additionally (i.e. add 

```js
          if (!uri.startsWith('mongodb://') && !uri.startsWith('mongodb+srv://')) {
            return 'MongoDB connection strings begin with "mongodb://" or "mongodb+srv://"';
```

before the `new ConnectionString` check), I think that’s okay and won’t make anything worse, but I also honestly think that this isn’t something we need to worry about a ton, and I do agree with @Anemy that it would be a bit nicer to improve the error message inside the connection string url package as a long-term solution :slightly_smiling_face: